packages/template-retail-react-app/app/hooks/use-item-movement.js

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) 2025, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useShopperBasketsMutation} from '@salesforce/commerce-sdk-react'
+import {useCallback} from 'react'
+import {useErrorHandler} from '@salesforce/retail-react-app/app/hooks/use-error-handler'
+import {DEFAULT_SHIPMENT_ID} from '@salesforce/retail-react-app/app/constants'
+
+/**
+ * Hook for item movement operations
+ * Focused only on moving items between shipments (basket data manipulation)
+ * @param {string} basketId - The basket ID
+ * @returns {Object} Object containing item movement functions
+ */
+export const useItemMovement = (basketId) => {
+    const handleError = useErrorHandler()
+    const updateItemInBasketMutation = useShopperBasketsMutation('updateItemInBasket')
+    const updateItemsInBasketMutation = useShopperBasketsMutation('updateItemsInBasket')
+
+    /**
+     * Moves a product item to a pickup shipment
+     * @param {Object} productItem - The product item to move
+     * @param {string} targetShipmentId - The target shipment ID
+     * @param {string} inventoryId - The inventory ID for the store
+     * @returns {Promise<Object>} The updated basket response
+     */
+    const moveItemToPickupShipment = useCallback(
```

Comment:
Since you are calling api for these func, let's use `update`
```suggestion
    const updateItemToPickupShipment = useCallback(
```